packages/core/lib/Interfaces/Context.ts

```diff
@@ -0,0 +1,34 @@
+import { IUser } from './User';
+
+export interface IContext {
+  tags?: any;
+  extra?: any;
+  user?: IUser;
+}
+
+export function getDefaultContext() {
+  return {};
+}
+
+export function set<T extends IContext, K extends keyof T>(
+  context: T,
+  key: K,
+  value: any
+) {
+  context[key] = value;
+}
+
+export function merge<T extends IContext, K extends keyof T>(
```

Comment:
Just a suggestion: I'd call it `mergeIn`, as `merge` usually refers to shallow-merging all keys of two or more objects.